OpenRA.Mods.Common/Activities/Air/FlyOffMap.cs

```diff
@@ -11,16 +11,43 @@
 
 using OpenRA.Activities;
 using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
 
 namespace OpenRA.Mods.Common.Activities
 {
 	public class FlyOffMap : Activity
 	{
 		readonly Aircraft aircraft;
+		readonly Target target;
+		readonly bool hasTarget;
 
 		public FlyOffMap(Actor self)
 		{
 			aircraft = self.Trait<Aircraft>();
+			ChildHasPriority = false;
+		}
+
+		public FlyOffMap(Actor self, Target target)
+		{
+			aircraft = self.Trait<Aircraft>();
+			ChildHasPriority = false;
+			this.target = target;
+			hasTarget = true;
+		}
+
+		protected override void OnFirstRun(Actor self)
+		{
+			if (hasTarget)
+			{
+				QueueChild(new Fly(self, target));
+				return;
+			}
+
+			// VTOLs must take off first if they're not at cruise altitude
+			if (aircraft.Info.VTOL && self.World.Map.DistanceAboveTerrain(aircraft.CenterPosition) != aircraft.Info.CruiseAltitude)
+				QueueChild(new TakeOff(self));
+
+			QueueChild(new FlyForward(aircraft));
```

Comment:
Instead of creating a new activity for this, couldn't we just call `FlyTimed` here with `ticks = -1`?

Reply:
Good idea.